client/opentracing/src/main/java/org/hawkular/apm/client/opentracing/DefaultNodeProcessor.java

```diff
@@ -45,7 +45,9 @@ public void process(TraceContext context, APMSpan span, NodeBuilder nodeBuilder)
                     } catch (MalformedURLException e) {
                         nodeBuilder.setUri(entry.getValue().toString());
                     }
-                    nodeBuilder.setEndpointType(entry.getKey().substring(0, entry.getKey().length() - 4));
+                    String type=entry.getKey().substring(0, entry.getKey().length() - 4);
```

Comment:
perhaps a if (entry.getKey.length() >=4) validation somewhere ?
But just if it makes sense.